agent/auto-config/config.go

```diff
@@ -49,14 +78,78 @@ type Config struct {
 	// having the test take minutes/hours to complete.
 	Waiter *lib.RetryWaiter
 
-	// CertMonitor is the Connect TLS Certificate Monitor to be used for ongoing
-	// certificate renewals and connect CA roots updates. This field is not
-	// strictly required but if not provided the TLS certificates retrieved
-	// through by the AutoConfig.InitialConfiguration RPC will not be used
-	// or renewed.
-	CertMonitor CertMonitor
-
 	// Loader merges source with the existing FileSources and returns the complete
 	// RuntimeConfig.
 	Loader func(source config.Source) (cfg *config.RuntimeConfig, warnings []string, err error)
+
+	// TLSConfigurator is the shared TLS Configurator. AutoConfig will update the
+	// auto encrypt/auto config certs as they are renewed.
+	TLSConfigurator TLSConfigurator
+
+	// Cache is an object implementing our Cache interface. The Cache
+	// used at runtime must be able to handle Roots and Leaf Cert watches
+	Cache Cache
+
+	// FallbackLeeway is the amount of time after certificate expiration before
+	// invoking the fallback routine. If not set this will default to 10s.
+	FallbackLeeway time.Duration
+
+	// FallbackRetry is the duration between Fallback invocations when the configured
+	// fallback routine returns an error. If not set this will default to 1m.
+	FallbackRetry time.Duration
+
+	// Tokens is the shared token store. It is used to retrieve the current
+	// agent token as well as getting notifications when that token is updated.
+	// This field is required.
+	Tokens TokenStore
+}
+
+// WithLogger will cause the created AutoConfig type to use the provided logger
+func (c *Config) WithLogger(logger hclog.Logger) *Config {
```

Comment:
I think these `WithX` builder functions are no longer used since we populate the `Config struct directly. Do you think they will be needed or could be remove them? Maybe this came from resolving a merge conflcit?

Reply:
I think we can remove them. They aren't used any more so we should get rid of the dead code.

Reply:
They have been removed.